environment.yml

```diff
@@ -17,3 +17,6 @@ dependencies:
   - pytest
   - pytest-cov
   - pre-commit
+  - pip:
+    # Optional dependency for the single-step model interface
+    - pydantic>=1.10.5
```

Comment:
I know this version requirement is copied in the internal repo, but is there any particular reason why it is required? Would anything bad happen if we unpinned it?

Reply:
Yeah, `1.10.2` has a bug related to `default_factory` (see [here](https://github.com/pydantic/pydantic/issues/5065)), so we wouldn't go to `<1.10` (we could maybe move the pin a few patch versions back, not sure exactly where between `1.10.2` and `1.10.5` the bug was fixed, but it doesn't seem worthwhile).